core/src/main/java/org/apache/calcite/sql/dialect/RedshiftSqlDialect.java

```diff
@@ -45,4 +49,24 @@ public RedshiftSqlDialect(Context context) {
       @Nullable SqlNode fetch) {
     unparseFetchUsingLimit(writer, offset, fetch);
   }
+
+  @Override public SqlNode getCastSpec(RelDataType type) {
+    String castSpec;
+    switch (type.getSqlTypeName()) {
```

Comment:
Since you are working on this It might worth checking if the rest of the types (at least the most common) in the `SqlTypeName` have an alternative for Redshift so that you fix everything in one-shot.

Reply:
Done. Found smallint (int2) and double precision (float8) that required attention. Fixed them for now in this patch and wrote the necessary test cases.